src/components/forms/Form.js

```diff
@@ -0,0 +1,450 @@
+ * 					type: "single"
+ * 				}
+ * 			}
+ * 		]
+ *
+ * 		initialValues = {
+ * 				"firstName": "Steven",
+ * 				"lastName": "Perdomo",
+ * 				"birthday": "2001-01-01",
+ * 				"coolnessLevel": "Super Uncool",
+ * 				"major": "Computer Science"
+ * 		}
+ *
+ * 		<Form
+ * 			elements = { elements }
+ * 			values = { values }
+ * 			title = "The Title"
+ * 			visible = { this.state.formVisibility }
+ * 			changeVisibility = { visible => this.setState({ formVisibility: visible }) }
+ * 			onSubmit = { value => console.log(`this is the value: ${value}`) }
+ * 			submitButtonName = "Confirm Coolness Level"
+ * 		/>
+ *
+ * 	Output:
+ * 		this.state = {
+ * 			firstName: "Haniel",
+ * 			lastName: "Diaz",
+ * 			birthday: null || "6969/04/20",
+ * 			coolnessLevel:  "Super Cool",
+ * 			major: "Computer Science"
+ * 		}
+ * 		onSubmit(this.state)
+ *
+ */
+
+class Form extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = this.props.values || {};
+	}
+
+	static propTypes = {
+		elements: PropTypes.arrayOf(
+			PropTypes.shape(
+				{
+					placeholder: PropTypes.string.isRequired,
+					camelCaseName: PropTypes.string.isRequired,
+					type: PropTypes.string.isRequired,
+					isRequired: PropTypes.bool,
+					isValid: PropTypes.func,
+					validShape: PropTypes.string,
+					options: PropTypes.any
+				}
+			)
+		).isRequired,
+		values: PropTypes.object,
+		title: PropTypes.string,
+		onSubmit: PropTypes.func,
+		onChange: PropTypes.func,
+		visible: PropTypes.bool.isRequired,
+		changeVisibility: PropTypes.func,
+		submitButtonName: PropTypes.string
+	}
+
+	componentDidUpdate(prevProps) {
+		if (!_.isEqual(prevProps.values, this.props.values)) this.setState(this.props.values);
+	}
+
+	/**
+	 * @description Processes an element's new value and updates the state of the form.
+	 *
+	 * If the element's value affects another element, the conditional values are applied to the form state.
+	 * The onChange prop is called here and sent the updated state.
+	 *
+	 * @param {Element} element The element whose value is being modified.
+	 * @param {Any} newValue The new value that will update the state.
+	 */
+
+	changeState(element, newValue) {
+		const { conditionalValues, camelCaseName } = element;
+		const updatedElement = { [camelCaseName]: newValue };
+
+		if (conditionalValues) this.applyConditionalValues(conditionalValues, newValue);
+		if (this.props.onChange) this.props.onChange(Object.assign(this.state, updatedElement));
+
+		this.setState(updatedElement);
+	}
+
+	/**
+	 * @description Uses a parent element's value to change the values of other elements in the form.
+	 *
+	 * If an element has nest elements inside of options.elements they will be included.
+	 *
+	 * @param {ConditionalValue[]} conditionalValues The conditionalValues of the parent element whose value determines the values of other form elements.
+	 * @param {Any} parentValue The new value of the parent element.
```

Comment:
Align correctly with min 2 spaces and separate `conditionalValues` with correct spaces